src/main/java/com/example/WeatherDataCodeChallenge/Sensor.java

```diff
@@ -1,25 +1,25 @@
 package com.example.WeatherDataCodeChallenge;
 
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
 import lombok.Data;
 import org.springframework.data.annotation.Id;
 import org.springframework.data.mongodb.core.mapping.Document;
-
 import javax.persistence.Embedded;
 
 @Data
 @Document
 public class Sensor {
 
-    @Id
```

Comment:
Why did you get rid of this annotation? You want your DB to know what the primary key is so it can find the object and update it easily